sdks/python/apache_beam/io/fileio.py

```diff
@@ -525,8 +526,11 @@ def __init__(
   @staticmethod
   def _get_sink_fn(input_sink):
     # type: (...) -> Callable[[Any], FileSink]
-    if isinstance(input_sink, FileSink):
-      return lambda x: input_sink
+    if isinstance(input_sink, type) and issubclass(input_sink, FileSink):
```

Comment:
For my learning, so this will work with:
sink=fileio.TextSink()
and
sink=fileio.TextSink

and previously it was only working with:
sink=fileio.TextSink

Is that correct?

Reply:
Yes this is correct. Previously sink=fileio.TextSink() generated unexpected behavior as stated in the ticket.

Reply:
How does that work even now, the call is:

sink = self.sink_fn(destination)

and all these lambdas are ignoring the input argument anyway.

Reply:
Now the call of `self.sink_fn(destination)` will produce a new FileSink instance and assign to sink. Yeah it sounds odd but FileSink constructors does not have any parameter passed in. The destination is actually assigned by the [open()](https://github.com/apache/beam/blob/d0c0dee32cf42fdaeb4e9b1556419906eb104029/sdks/python/apache_beam/io/fileio.py#L812) method.

Reply:
Thank you for the explanation, and sorry for the repeated questions. One more question, what worked before?

I guess the only thing that worked before was either passing callable or None. Was there anyway to pass FileSink or FileSink() or any other variation here?

Reply:
> I guess the only thing that worked before was either passing callable or None. 

Yes exactly; 

> Was there anyway to pass FileSink or FileSink() or any other variation here?

Passing FileSink() could work if there was only one destination.